### ISSUE TITLE
CA1822 does not trigger if a method uses an object/collection initializer (can't reproduce via unit tests?)

### ISSUE THREAD

**Issue by rwdalpe** [dotnet/roslyn-analyzers#2005]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/)

#### Package Version

v2.6.2 and v2.6.3

#### Diagnostic ID

[CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822-mark-members-as-static?view=vs-2017)

#### Repro steps

* Create a new .NET Framework or .NET Core project in Visual Studio.
* Add Microsoft.CodeQuality.Analyzers as a nuget package to the project. 
* Set the severity of CA1822 to Error.
* Add a new file to the project containing this content:
```
using System.Collections.Generic;

public class CA1822Test
{
    public IEnumerable<int> MethodWithCollectionInitializer()
    {
        return new List<int>() { 1 };
    }

    public IEnumerable<int> MethodWithoutCollectionInitializer()
    {
        return new List<int>();
    }
}
```
* Rebuild the project.

#### Expected behavior

Both `MethodWithCollectionInitializer` and `MethodWithoutCollectionInitializer` trigger an error on CA1822.

#### Actual behavior

Only `MethodWithoutCollectionInitializer` triggers an error on CA1822.

#### Other Information

Something strange about this is that I tried adding code like `MethodWithCollectionInitializer` to the unit tests for CA1822 in `MarkMembersAsStaticTests` and it triggered the diagnostic error (both on `master` and the commit tagged for `2.6.2`). I can only reproduce this issue in an actual project.

I think this is ultimately caused by https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs#L79

Per the [documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.operations.iinstancereferenceoperation?view=roslyn-dotnet) `OperationKind.InstanceReference` can refer to 

>(1) C# this or base expression. (2) VB Me, MyClass, or MyBase expression. (3) C# object or collection initializers. (4) VB With statements, object or collection initializers.

There is a `ReferenceKind` property on the operation that might be able to be used to differentiate between initializers and other usages. I'm definitely not an expert, but I'm not aware of a reason why we would skip triggering CA1822 just because an initializer was used in the method.

**Comment by mavasani** [comment]
Fixed in 2.6.3 release with https://github.com/dotnet/roslyn-analyzers/pull/1976

**Comment by mavasani** [comment]
Let me retry the repro and get back to you.

**Comment by rwdalpe** [comment]
@mavasani just for my own information, is there a reason why this cannot be reproduced in the unit tests for the CA1822 analyzer? I tried adding the troublesome code on both `master` and the commit associated with the `2.6.2` release and the analyzer was always triggered during the test.
